packages/venia-concept/src/components/SearchBar/searchBar.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Route } from 'react-router-dom';
+import { withRouter } from 'react-router';
+
+import Icon from 'src/components/Icon';
+import SeedSearchInput from './seedSearchInput';
+
+import classify from 'src/classify';
+import defaultClasses from './searchBar.css';
+
+export class SearchBar extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            clearIcon: PropTypes.string,
+            clearIcon_off: PropTypes.string,
+            root: PropTypes.string,
+            searchBlock: PropTypes.string,
+            searchBlockOpen: PropTypes.string,
+            searchBar: PropTypes.string,
+            searchIcon: PropTypes.string
+        }),
+        executeSearch: PropTypes.func.isRequired,
+        history: PropTypes.object,
+        isOpen: PropTypes.bool,
+        location: PropTypes.object,
+        match: PropTypes.object
+    };
+
+    constructor(props) {
+        super(props);
+        this.searchRef = React.createRef();
+        this.state = { showClearIcon: false };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.isOpen !== prevProps.isOpen) {
+            if (this.props.isOpen == true) {
+                this.searchRef.current.focus();
+            } else {
+                this.searchRef.current.blur();
+            }
+        }
+    }
+
+    enterSearch = event => {
+        const searchQuery = this.searchRef.current.value;
+        this.setClearIcon(searchQuery);
+
+        if (
+            this.props.isOpen &&
+            (event.type === 'click' || event.key === 'Enter') &&
+            searchQuery !== ''
+        ) {
+            this.props.executeSearch(searchQuery, this.props.history);
+        }
+    };
+
+    clearSearch = () => {
+        this.searchRef.current.value = '';
```

Comment:
Likewise, this is not how we control inputs in React. We don't write to the DOM—React does.

Here, `value` is a piece of state; whatever that state is should be applied to the input  during `render()`. The role of `clearSearch()` should only be to set the `value` state to `""`.